### PR TITLE
Update countdown_timer.jsx

### PR DESCRIPTION
**Pull request by vaibhavn29**
Should replay timer flag added

### CODE REVIEW

File: countdown_timer.jsx [hunk 1]
```diff
@@ -45,8 +47,11 @@ var CountdownTimer = React.createClass({
   },
 
   componentWillReceiveProps: function(newProps, oldProps) {
-    if (this.state.timeoutId) clearTimeout(this.state.timeoutId);
-    this.setState({ prevTime: undefined, timeRemaining: newProps.initialTimeRemaining });
+    //shouldReplayTimer is false if you want to run the counter only once
+    if(this.props.shouldReplayTimer){
+      if (this.state.timeoutId) clearTimeout(this.state.timeoutId);
+      this.setState({ prevTime: undefined, timeRemaining: newProps.initialTimeRemaining });
```
Comment (rvierich) [inline countdown_timer.jsx:51]: Updated master to use `null` instead of `undefined` for state defaults.

**Comment by rvierich** [comment]
Inside `tick()`, there's a check to determine if the timer is complete. I think that's the best spot to add the check for `shouldReplayTimer`, and reset it to `this.props.initialTimeRemaining` at that point.

Where it is currently (`componentWillRecieveProps`), it's functionality is effectively: "if I get re-rendered with new props, and `shouldReplayTimer` was set to true on my **previous** render (because we're checking `this.props.shouldReplayTimer` instead of `newProps.shouldReplayTimer`), then restart the timer using the new props."

**Comment by rvierich** [comment]
@overthinker29 Do you plan on updating this?

**Comment by vaibhavn29** [comment]
Hey thanks for the comments and suggestions.
Let me know if the updated code makes sense...

File: countdown_timer.jsx [hunk 2]
```diff
@@ -91,6 +93,7 @@ var CountdownTimer = React.createClass({
 
     if (countdownComplete) {
       if (this.props.completeCallback) { this.props.completeCallback() };
+      if(this.props.shouldReplayTimer){timeRemaining=this.props.initialTimeRemaining};
```
Comment (rvierich) [inline countdown_timer.jsx:96]: This won't actually update the component's state. You need to use `this.setState()` to update `this.state.timeRemaining`.

See: https://facebook.github.io/react/docs/tutorial.html#reactive-state

**Comment by rvierich** [comment]
@overthinker29 Looks better! I added one more comment though.

**Comment by rvierich** [comment]
Closing as this has been open for a while without updates.

**Pull request closed without merging**